File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        evenPointer = 0
        oddPointer = 1
        while( evenPointer < len(nums) and oddPointer < len(nums)):
            if(not nums[evenPointer]%2):
                evenPointer+=2
            elif(nums[oddPointer]%2):
                oddPointer+=2
            else:
                temp = nums[evenPointer]
                nums[evenPointer] = nums[oddPointer]
                nums[oddPointer] = temp
        return nums

File: 0922-sort-array-by-parity-ii/0922-sort-array-by-parity-ii.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArrayByParityII = function(nums) {
        let evenPointer = 0;
        let oddPointer = 1;
        while(evenPointer < nums.length && oddPointer < nums.length){
            if(nums[evenPointer]%2 == 0) evenPointer+=2;
            else if(nums[oddPointer]%2 == 1) oddPointer+=2;
            else {
                let temp = nums[evenPointer];
                nums[evenPointer] = nums[oddPointer];
                nums[oddPointer] = temp;
            }
        }
        return nums;  
};